src/wasm/kotlin/src/arcs/Particle.kt

```diff
@@ -83,10 +83,10 @@ abstract class Particle : WasmObject() {
      * @return absolute URL
      */
     fun resolveUrl(url: String): String {
-      val r: WasmString = resolveUrl(url.toWasmString())
-      val resolved = r.toKString()
-      _free(r)
-      return resolved
+        val r: WasmString = resolveUrl(url.toWasmString())
+        val resolved = r.toKString()
+        _free(r)
```

Comment:
can we create an extension function for `WasmString`?

```kotlin
fun WasmString.free() = _free(this)
```

Reply:
Also, maybe this whole method could be an extension function on `String` outside of the particle itself? It doesn't seem to rely on any instance variables.

Reply:
That seems odd to me - `resolveUrl` is specific to URL fields (which happen to map to the string type), not strings in general. It also seems like a code smell to update a core data type with application-specific logic (just like extending native JS object prototypes is usually considered bad practice).

Reply:
Extending native JS prototypes actually modifies the underlying class for anything in the runtime, extension functions compiled into static methods.  So unless you're importing those extension functions, they don't exist on the class.

Reply:
Nice, that removes the second point, but I still think `resolveUrl` should only be applicable to URL fields. I gather using `typealias URL = String` wouldn't fix this - looks like extension functions on URL would be usable on Strings as well (as you'd expect).